### PR TITLE
Update OpenAPI Planner to use params and shared memory

### PR DESCRIPTION
**Pull request by ar-mccabe**
The main change proposed here is to introduce params as a third argument for the class `RequestsGetToolWithParsing` as well as the corresponding instruction prompt, `REQUESTS_GET_TOOL_DESCRIPTION`. 

I've tested this on a fork and it appears to be working. Previously I had observed behaviour in which the Agent would attempt to return/list all results then follow up with a filter or truncation (when a GET list endpoint is available). However, due to the `RESPONSE_LENGTH`, any paginated or long responses perform poorly in this approach. Therefore, adding the params query allows the Agent to default to params when available (and `None` if not applicable).

Additionally, I have experimented with shared memory for the planner Agent. It's unclear if it is having the desired effect, however it doesn't throw errors and in theory should give the Planner more context on the entire conversation.

### CODE REVIEW

File: langchain/agents/agent_toolkits/openapi/planner.py [hunk 1]
```diff
@@ -185,6 +190,7 @@ def create_openapi_agent(
     api_spec: ReducedOpenAPISpec,
     requests_wrapper: RequestsWrapper,
     llm: BaseLanguageModel,
+    shared_memory: ReadOnlySharedMemory,
```
Comment (hwchase17) [inline langchain/agents/agent_toolkits/openapi/planner.py:193]: lets maybe make this optional, i dont think always needed

**Review by hwchase17 — COMMENTED** [review]
looks pretty good to me, @jerwelborn any thoughts?

File: langchain/agents/agent_toolkits/openapi/planner.py (hunk 1 above)
Comment (ar-mccabe) [reply langchain/agents/agent_toolkits/openapi/planner.py:193]: Good point; I'll update

Comment (ar-mccabe) [reply langchain/agents/agent_toolkits/openapi/planner.py]: Updated to optional in https://github.com/hwchase17/langchain/pull/2475/commits/678a40d1d8d1ee40219ac7474f81b7f2c9ecd3b4

**Comment by ar-mccabe** [comment]
https://github.com/hwchase17/langchain/pull/2475/commits/ba011bc6b7322b7af66bd474dd8d6e6943c8eea9 Fixed the failed linting test introduced with the missed Optional wrapper on ReadOnlySharedMemory type.

**Comment by ar-mccabe** [comment]
Had some style issues, https://github.com/hwchase17/langchain/pull/2475/commits/704ce15da7d889490d08c75c545364e34b5eced2 should fix that.

**Comment by ar-mccabe** [comment]
Again, hoping https://github.com/hwchase17/langchain/pull/2475/commits/f0d671e9684ecf4849e6094baa433b36d31ac17c fixes the most recent linting error. First time using this linter so apologies for the repeated failed jobs.

File: langchain/agents/agent_toolkits/openapi/planner.py [hunk 2]
```diff
@@ -55,7 +56,11 @@ def _run(self, text: str) -> str:
             data = json.loads(text)
         except json.JSONDecodeError as e:
             raise e
-        response = self.requests_wrapper.get(data["url"])
+        try:
+            data_params = data["params"]
```
Comment (vowelparrot) [inline langchain/agents/agent_toolkits/openapi/planner.py:60]: Preferred:
```
data_params = data.get("params")
```

over the try/catch block

**Pull request merged**